src/test/java/org/jenkinsci/plugins/docker/workflow/ServerEndpointStepTest.java

```diff
@@ -101,4 +111,58 @@
         });
     }
 
+    @Test public void stepExecutionWithCredentials() {
+        story.addStep(new Statement() {
+            @Override public void evaluate() throws Throwable {
+                assumeNotWindows();
+                IdCredentials serverCredentials = new DockerServerCredentials(CredentialsScope.GLOBAL, "serverCreds", null, "clientKey", "clientCertificate", "serverCaCertificate");
+                CredentialsProvider.lookupStores(story.j.jenkins).iterator().next().addCredentials(Domain.global(), serverCredentials);
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "prj");
+                p.setDefinition(new CpsFlowDefinition(
+                        "node {\n" +
+                                "  withDockerServer(server: [uri: 'tcp://host:1234', credentialsId: 'serverCreds']) {\n" +
+                                "    sh 'echo would be connecting to $DOCKER_HOST'\n" +
+                                "  }\n" +
+                                "}", true));
+                WorkflowRun b = story.j.buildAndAssertSuccess(p);
+                story.j.assertLogContains("would be connecting to tcp://host:1234", b);
```

Comment:
But you are not actually testing the credentials here. What then happens with `src/main/` reverted? Is there a build failure, or are the credentials ignored, or a warning logged?

Reply:
You're right. I am going to add test on `DOCKER_TLS_VERIFY` and `DOCKER_CERT_PATH` that should be injected by the step. I have noticed that in the case of `JENKINS-48437`, there is not failure but those variables are not filled.